### DISCUSSION TITLE
more duck control

### DISCUSSION THREAD

**Discussion by 733737** [OpenRCT2/OpenRCT2#17604] [Feature requests]
fidwell wants to make a plugin for ducks, is there a possibility the devs could make the spawn rate and quantity of ducks accessible to the plugin api?

**Comment by IntelOrca** [comment]
Would just the ability to create new ducks and hook the existing in-game create duck call be enough?

**Reply by fidwell** [reply]
I only mentioned this in passing and have put zero actual thought into it. But yeah, I just wanted a way to automatically spawn them, so I suppose merely the ability to create new ducks is sufficient. I'm not sure what I would use a hook to the "create duck" action for.

**Reply by IntelOrca** [reply]
You need to hook to disable the original spawn behaviour.

**Reply by fidwell** [reply]
Oh gotcha, that makes sense.

**Reply by fidwell** [reply]
Actually, if the [randomizer plugin](https://github.com/Die4Ever/rollercoaster-tycoon-randomizer/search?q=ducks) is to be believed, spawning ducks is already handled with the `setCheatAction` action? I should test that out.

**Comment by fidwell** [comment]
Yeah turns out this is already easily done.

![image](https://user-images.githubusercontent.com/5436387/181865760-37493709-406d-4f78-9b7f-d1223388e790.png)

**Reply by fidwell** [reply]
Discovered an issue with this, though—since it uses the cheat action, in multiplayer, every player must have cheat permissions or else they'll get a permissions error every time the subscribed event fires. I don't always give other players cheat permissions. It still works, but it's not a good idea to annoy the other players.
